### ISSUE TITLE
BinaryPoly division bug

### ISSUE THREAD

**Issue by mhostetter** [mhostetter/galois#113]
The division works correctly using the `DensePoly` implementation but not the `BinaryPoly` implementation.

```python
In [15]: a = galois.Poly.String("x^128"); b = galois.Poly.String("x^105 + x^104 + x^103 + x^102 
    ...: + x^101 + x^100 + x^99 + x^98 + x^97 + x^96 + x^95 + x^94 + x^93 + x^92 + x^91 + x^90 +
    ...:  x^89 + x^88 + x^87 + x^86 + x^85 + x^84 + x^83 + x^82 + x^81 + x^80 + x^79 + x^78 + x^
    ...: 77 + x^76 + x^75 + x^74 + x^73 + x^72 + x^71 + x^70 + x^69 + x^68 + x^67 + x^66 + x^65 
    ...: + x^64 + x^63 + x^62 + x^61 + x^60 + x^59 + x^58 + x^57 + x^56 + x^55 + x^54 + x^53 + x
    ...: ^52 + x^51 + x^50 + x^49 + x^48 + x^47 + x^46 + x^45 + x^44 + x^43 + x^42 + x^41 + x^40
    ...:  + x^39 + x^38 + x^37 + x^36 + x^35 + x^34 + x^33 + x^32 + x^31 + x^30 + x^29 + x^28 + 
    ...: x^27 + x^26 + x^25 + x^24 + x^23 + x^22 + x^21 + x^20 + x^19 + x^18 + x^17 + x^16 + x^1
    ...: 5 + x^14 + x^13 + x^12 + x^11 + x^10 + x^9 + x^8 + x^7 + x^6 + x^5 + x^4 + x^3 + x^2 + 
    ...: x")                                                                                    

In [16]: a                                                                                      
Out[16]: Poly(x^128, GF(2))

In [17]: b                                                                                      
Out[17]: Poly(x^106 + x^105 + x^104 + x^103 + x^102 + x^101 + x^100 + x^99 + x^98 + x^97 + x^96 + x^95 + x^94 + x^93 + x^92 + x^91 + x^90 + x^89 + x^88 + x^87 + x^86 + x^85 + x^84 + x^83 + x^82 + x^81 + x^80 + x^79 + x^78 + x^77 + x^76 + x^75 + x^74 + x^73 + x^72 + x^71 + x^70 + x^69 + x^68 + x^67 + x^66 + x^65 + x^64 + x^63 + x^62 + x^61 + x^60 + x^59 + x^58 + x^57 + x^56 + x^55 + x^54 + x^53 + x^52 + x^51 + x^50 + x^49 + x^48 + x^47 + x^46 + x^45 + x^44 + x^43 + x^42 + x^41 + x^40 + x^39 + x^38 + x^37 + x^36 + x^35 + x^34 + x^33 + x^32 + x^31 + x^30 + x^29 + x^28 + x^27 + x^26 + x^25 + x^24 + x^23 + x^22 + x^21 + x^20 + x^19 + x^18 + x^17 + x^16 + x^15 + x^14 + x^13 + x^12 + x^11 + x^10 + x^9 + x^8 + x^7 + x^6 + x^5 + x^4 + x^3 + x^2, GF(2))

In [18]: a / b                                                                                  
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-18-d8e10abd5ab6> in <module>
----> 1 a / b

~/repos/galois/galois/field/poly.py in __truediv__(self, other)
    842     def __truediv__(self, other):
    843         cls, a, b = self._check_inputs_are_polys(self, other)
--> 844         return cls._divmod(a, b)[0]
    845 
    846     def __rtruediv__(self, other):

~/repos/galois/galois/field/poly.py in _divmod(cls, a, b)
   1255                 a ^= b << i
   1256                 q ^= 1  # Set the LSB then left shift
-> 1257             assert a & mask == 0
   1258             mask >>= 1
   1259 

AssertionError: 

In [19]: q, r = galois.field.poly.DensePoly._divmod(a, b); q, r                                 
Out[19]: (Poly(x^23 + x^22, GF(2)), Poly(x^23, GF(2)))

In [20]: q*b + r, a                                                                             
Out[20]: (Poly(x^128, GF(2)), Poly(x^128, GF(2)))
```

**Comment by mhostetter** [comment]
Again `math.log` and `math.log2` were causing problems for large integers.

```python
    @property
    def degree(self):
        if self._integer == 0:
            return 0
        else:
            return int(math.floor(math.log2(self._integer)))
```

needed to be replaced with:

```python
    @property
    def degree(self):
        if self._integer == 0:
            return 0
        else:
            return len(bin(self._integer)[2:]) - 1
```

then the division works as expected

```python
In [1]: import galois                                                                           

In [2]: a = galois.Poly.String("x^128"); b = galois.Poly.String("x^105 + x^104 + x^103 + x^102 +
   ...:  x^101 + x^100 + x^99 + x^98 + x^97 + x^96 + x^95 + x^94 + x^93 + x^92 + x^91 + x^90 + x
   ...: ^89 + x^88 + x^87 + x^86 + x^85 + x^84 + x^83 + x^82 + x^81 + x^80 + x^79 + x^78 + x^77 
   ...: + x^76 + x^75 + x^74 + x^73 + x^72 + x^71 + x^70 + x^69 + x^68 + x^67 + x^66 + x^65 + x^
   ...: 64 + x^63 + x^62 + x^61 + x^60 + x^59 + x^58 + x^57 + x^56 + x^55 + x^54 + x^53 + x^52 +
   ...:  x^51 + x^50 + x^49 + x^48 + x^47 + x^46 + x^45 + x^44 + x^43 + x^42 + x^41 + x^40 + x^3
   ...: 9 + x^38 + x^37 + x^36 + x^35 + x^34 + x^33 + x^32 + x^31 + x^30 + x^29 + x^28 + x^27 + 
   ...: x^26 + x^25 + x^24 + x^23 + x^22 + x^21 + x^20 + x^19 + x^18 + x^17 + x^16 + x^15 + x^14
   ...:  + x^13 + x^12 + x^11 + x^10 + x^9 + x^8 + x^7 + x^6 + x^5 + x^4 + x^3 + x^2 + x")      

In [3]: divmod(a, b)                                                                            
Out[3]: (Poly(x^23 + x^22, GF(2)), Poly(x^23, GF(2)))
```

**Comment by mhostetter** [comment]
Implemented in ee66cc2d66cefc5c2cde5078604e25ba5c191985.
